template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/spark/SparkOnYarnRoleConfigProvider.java

```diff
@@ -0,0 +1,55 @@
+package com.sequenceiq.cloudbreak.cmtemplate.configproviders.spark;
+
+import static com.sequenceiq.cloudbreak.cmtemplate.configproviders.ConfigUtils.config;
+import static com.sequenceiq.cloudbreak.cmtemplate.configproviders.hive.HiveMetastoreCloudStorageServiceConfigProvider.HMS_METASTORE_EXTERNAL_DIR;
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+import org.springframework.stereotype.Component;
+
+import com.cloudera.api.swagger.model.ApiClusterTemplateConfig;
+import com.sequenceiq.cloudbreak.cmtemplate.CmTemplateProcessor;
+import com.sequenceiq.cloudbreak.cmtemplate.configproviders.AbstractRoleConfigProvider;
+import com.sequenceiq.cloudbreak.cmtemplate.configproviders.ConfigUtils;
+import com.sequenceiq.cloudbreak.template.TemplatePreparationObject;
+
+@Component
+public class SparkOnYarnRoleConfigProvider extends AbstractRoleConfigProvider {
+
+    private static final String SPARK_CONF_CLIENT_SAFETY_VALVE = "spark-conf/spark-defaults.conf_client_config_safety_valve";
+
+    private static final String SPARK_YARN_ACCESS_DIR_PARAM = "spark.yarn.access.hadoopFileSystems=";
+
+    private static final Pattern BASE_PATH_PATTERN = Pattern.compile("([\\w]+)(?:/(.*))?");
```

Comment:
Would be nice if existing pattern/logic could be reused from:

https://github.com/hortonworks/cloudbreak/blob/50b0b2b8a831a6b14e827b324042ae33b50def78/template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/zeppelin/ZeppelinS3CloudStorageRoleConfigProvider.java#L31

https://github.com/hortonworks/cloudbreak/blob/50b0b2b8a831a6b14e827b324042ae33b50def78/template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/zeppelin/ZeppelinS3CloudStorageRoleConfigProvider.java#L68

Reply:
Thanks, have moved the pattern to ConfigUtils and have reused the BASE_PATH_PATTERN for deriving zeppelin config values.

Reply:
Thanks!